### ISSUE TITLE
[NamedScopes] NoMethodError (undefined method `to_amf' for #<Class:0x...>)

### ISSUE THREAD

**Issue by ximus** [dima/restfulx#20]
It's late for me and I can't test this further.

On the Rails (server) side, AMF seems to work great now except I get:

NoMethodError (undefined method `to_amf' for #Class:0x...)

or

NoMethodError (undefined method `to_fxml' for #Class:0x...)

when I try to render results from a named scope.
I have jumped in with the debugger and:

@results.class ouputs ActiveRecord::NamedScope::Scope

not sure why the error says its an instance of Class and not ActiveRecord::NamedScope::Scope but thats probably a limitation of my knowledge of ruby.

Seems the AMF and FXML foo isn't getting to ActiveRecord::NamedScope::Scope

**Comment by ximus** [comment]
seems support for name scopes was taken out in commit d27de13735931a2e9aff510fe36158e637e5182a

The following was taken out from lib/restfulx.rb and, seemingly, never replaced:

module ActiveRecord
    # ActiveRecord named scopes are computed _before_ restfulx gem gets loaded
    # this patch addresses that and makes sure +to_fxml+ calls are properly
    # delegated
    module NamedScope
      # make sure we properly delegate +to_fxml+ calls to the proxy
      class Scope
        delegate :to_fxml, :to => :proxy_found
      end
    end
  end

**Comment by ximus** [comment]
I added this code to the current lib/restfulx/active_record.rb and this solved the issue.
Also I added a delegation for AMF support:
delegate :to_amf,  :to => :proxy_found
works fine, Dima up to you how you want to fix this.
Maybe a quick and not so dirty solution is viable considering this is going to be scrapped upon Rails3 support. But I don't know wether you plan on maintaining  Rails2.x.x support in parallel to Rails3.
